source/extensions/clusters/dynamic_forward_proxy/cluster.h

```diff
@@ -145,13 +173,29 @@ class Cluster : public Upstream::BaseDynamicClusterImpl,
   const envoy::config::endpoint::v3::LocalityLbEndpoints dummy_locality_lb_endpoint_;
   const envoy::config::endpoint::v3::LbEndpoint dummy_lb_endpoint_;
   const LocalInfo::LocalInfo& local_info_;
+  Event::Dispatcher& main_thread_dispatcher_;
+  const envoy::config::cluster::v3::Cluster orig_cluster_config_;
+
+  Event::TimerPtr idle_timer_;
 
   // True if H2 and H3 connections may be reused across different origins.
   const bool allow_coalesced_connections_;
 
   mutable absl::Mutex host_map_lock_;
   HostInfoMap host_map_ ABSL_GUARDED_BY(host_map_lock_);
 
+  mutable absl::Mutex cluster_map_lock_;
+  ClusterInfoMap cluster_map_ ABSL_GUARDED_BY(cluster_map_lock_);
+
+  // only for isShutdown.
```

Comment:
What does this mean?

Reply:
`only for using tls_.isShutdown() in Cluster::~Cluster`, also updated the comment.